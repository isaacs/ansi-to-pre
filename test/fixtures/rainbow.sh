#!/usr/bin/env bash
seq=$'\u2588\u2588\u2593\u2593\u2592\u2592\u2591\u2591'

r=(1 3 2 6 4 5)
rainbow () {
  f=$1
  b=$2
  t=$3
  echo -n $'\x1b['$t';30;100m'$seq
  for i in {0..6}; do
    fc=$([ $i -eq 0 ] && echo 90 || echo "$f${r[$[$i - 1]]}")
    bc=$([ $i -eq 6 ] && echo 100 || echo "$b${r[$i]}")
    echo -n $'\x1b['"${fc};${bc}m$seq"
  done
  echo -n $'\x1b[90;40m'$seq
  echo $'\x1b[40m  \x1b[0;22m'
}
rainbow 3 4 2
rainbow 3 4
rainbow 9 10
rainbow 9 10 2
