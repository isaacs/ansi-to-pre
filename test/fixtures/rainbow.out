[2;30;100mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘[90;41mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘[31;43mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘[33;42mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘[32;46mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘[36;44mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘[34;45mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘[35;100mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘[90;40mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘[40m  [0;22m
[;30;100mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘[90;41mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘[31;43mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘[33;42mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘[32;46mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘[36;44mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘[34;45mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘[35;100mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘[90;40mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘[40m  [0;22m
[;30;100mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘[90;101mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘[91;103mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘[93;102mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘[92;106mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘[96;104mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘[94;105mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘[95;100mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘[90;40mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘[40m  [0;22m
[2;30;100mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘[90;101mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘[91;103mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘[93;102mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘[92;106mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘[96;104mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘[94;105mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘[95;100mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘[90;40mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘[40m  [0;22m
